runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/http/compression/CompressionAlgorithm.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.compression
```

Comment:
**fix**: This probably belongs in the `http` module (package name is fine). Same thing for the `Gzip` implementation.`runtime-core` should just have the primitives for implementing the compression (so things like `GzipSdkSource` or `GzipByteReadChannel`, etc).